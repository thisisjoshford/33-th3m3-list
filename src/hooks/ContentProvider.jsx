import React, { createContext, useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { getPeople } from '../services/apiFetch';

export const XfilesContent = createContext();

export const ContentProvider = ({ children }) => {
  const [characters, setCharacters] = useState([]);

  useEffect(() => {
    getPeople()
      .then(peeps => setCharacters(peeps));
  }, []);

  return (
    <XfilesContent.Provider value={characters}>
      {children}
    </XfilesContent.Provider>
  );
};

ContentProvider.propTypes = {
  children: PropTypes.node
};

