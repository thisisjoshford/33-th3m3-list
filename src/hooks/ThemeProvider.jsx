import React, { createContext, useState, useContext } from 'react';
import PropTypes from 'prop-types';

const ThemeContext = createContext();

export const ThemeProvider = ({ children }) => {
  const [theme, setTheme] = useState('light');
  const toggle = ({ target }) => {
    if(target.checked) setTheme('dark');
    if(!target.checked) setTheme('light');
  };

  return (
    <ThemeContext.Provider value={{ theme, toggle }}>
      {children}
    </ThemeContext.Provider>
  );
};

ThemeProvider.propTypes = {
  children: PropTypes.node
};

export const useTheme = () => {
  const theme = useContext(ThemeContext);
  return theme;
};

export const useToggle = () => {
  const toggle = useContext(ThemeContext);
  return toggle;
};
