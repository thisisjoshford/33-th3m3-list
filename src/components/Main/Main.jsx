import React from 'react';
import Header from '../Header/Header';
import ListPage from '../ListPage/ListPage';
import { useTheme } from '../../hooks/ThemeProvider';
import ListStyle from '../ListPage/ListPage.css';

export default function Main() {
  const { theme } = useTheme();
  
  let styling;
  if(theme === 'light') styling = ListStyle.Light;
  if(theme === 'dark') styling = ListStyle.Dark;

  return (
    <div className={styling}>
      <Header />
      <ListPage />
    </div>
  );
}  
