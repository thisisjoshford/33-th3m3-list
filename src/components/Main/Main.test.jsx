import React from 'react';
import { shallow } from 'enzyme';
import Main from './Main';
import { ThemeProvider } from '../../hooks/ThemeProvider';

describe('Main component', () => {

  it('renders Main', () => {
    const wrapper = shallow(
      <ThemeProvider>
        <Main/>
      </ThemeProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
