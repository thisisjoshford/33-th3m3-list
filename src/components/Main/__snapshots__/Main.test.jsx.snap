// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component renders Main 1`] = `
<ContextProvider
  value={
    Object {
      "theme": "light",
      "toggle": [Function],
    }
  }
>
  <Main />
</ContextProvider>
`;
